lib/cn/csched_sp3_work.c

```diff
@@ -326,46 +326,82 @@ sp3_work_wtype_split(
     struct cn_tree *tree = tn->tn_tree;
     struct kvset_list_entry *le;
     struct list_head *head;
-    uint64_t now;
+    bool splittable;
     uint kvsets;
 
     head = &tn->tn_kvset_list;
     *mark = list_last_entry_or_null(head, typeof(*le), le_link);
-    kvsets = cn_ns_kvsets(&tn->tn_ns);
-
     *action = CN_ACTION_SPLIT;
     *rule = CN_RULE_SPLIT;
 
+    kvsets = cn_ns_kvsets(&tn->tn_ns);
+
+    /* Recheck to see if this node should be split, as while awaiting
+     * an opportunity to split an rspill could have shrunk this node
+     * below the split size.
+     */
+    splittable = (cn_ns_clen(&tn->tn_ns) >= tn->tn_split_size) ||
+        (cn_ns_keys_uniq(&tn->tn_ns) >= thresh->lcomp_split_keys);
+    splittable &= (kvsets > 0);
```

Comment:
`&&=` ?
Maybe make it a single statement:
```
splittable =  (kvsets > 0) && (....);
```

Reply:
Yeah, I meant to also add the following block to that check..

Reply:
nevermind, that's something from an upcoming patch..

Reply:
My main comment was about use of bit-wise AND `&=` instead of logical AND.

Reply:
My guess as to why there is no &&= or ||= is that &= and L= on booleans is sufficient?

Reply:
Ahh.  I didn't immediately grok that bit ops on Boolean types are the same as logical ops.  Still, seems risky to rely on that behavior in C where it is so common to use integers as Booleans.

Reply:
Yeah, I understand the concern.  The RHS is boolean now, but someone who doesn't grok it could break it easily...